File: applications_server/api/dto/field_of_study_dto.py
from api.dto.with_config import WithConfig


class FieldOfStudyDTO(WithConfig):
    field_of_study_id: int
    field_of_study_name: str


class FilteredFieldOfStudyDTO(WithConfig):
    fields_of_study: list[FieldOfStudyDTO]
    total_count: int

File: applicants_administration/src/entities/FieldOfStudy/models/fieldOfStudyDto.ts
export class FieldOfStudyDTO {
    field_of_study_id: number;
    field_of_study_name: string;

    constructor(id: number, name: string) {
        this.field_of_study_id = id;
        this.field_of_study_name = name;
    }
}

export class FieldOfStudyDTOSet {
    fields_of_study: FieldOfStudyDTO[];
    total_count: number;

    constructor(fields_of_study: FieldOfStudyDTO[], total_count: number) {
        this.fields_of_study = fields_of_study;
        this.total_count = total_count;
    }
}